plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,567 @@ let
+      current = helpers.defaultNullOpts.mkBool false ''
+        Controls the visibility of the current buffer.
+      '';
 
-    highlightVisible = helpers.defaultNullOpts.mkBool true "Highlight visible buffers";
+      extensions = helpers.defaultNullOpts.mkBool false ''
+        Controls the visibility of file extensions.
+      '';
 
-    icons =
-      stateOptions
-      // (mapAttrs
-        (
-          name: description:
-          mkOption {
-            type = types.submodule { options = stateOptions; };
-            default = { };
-            inherit description;
-          }
-        )
-        {
-          alternate = "The icons used for an alternate buffer.";
-          current = "The icons for the current buffer.";
-          inactive = "The icons for inactive buffers.";
-          visible = "The icons for visible buffers.";
-        }
-      );
+      inactive = helpers.defaultNullOpts.mkBool false ''
+        Controls visibility of `|hidden-buffer|`s and `|inactive-buffer|`s.
+      '';
 
-    hide = {
-      alternate = helpers.mkNullOrOption types.bool "Hide alternate buffers";
-      current = helpers.mkNullOrOption types.bool "Hide current buffer";
-      extensions = helpers.mkNullOrOption types.bool "Hide file extensions";
-      inactive = helpers.mkNullOrOption types.bool "Hide inactive buffers";
-      visible = helpers.mkNullOrOption types.bool "Hide visible buffers";
+      visible = helpers.defaultNullOpts.mkBool false ''
+        Controls visibility of `|active-buffer|`s.
+        `highlight_visible` must be `true`.
+      '';
     };
 
-    insertAtEnd = helpers.defaultNullOpts.mkBool false ''
-      If true, new buffers will be inserted at the end of the list.
-      Default is to insert after current buffer.
+    highlight_alternate = helpers.defaultNullOpts.mkBool false ''
+      Enables highlighting of alternate buffers.
     '';
 
-    insertAtStart = helpers.defaultNullOpts.mkBool false ''
-      If true, new buffers will be inserted at the start of the list.
-      Default is to insert after current buffer.
+    highlight_inactive_file_icons = helpers.defaultNullOpts.mkBool false ''
+      Enables highlighting the file icons of inactive buffers.
     '';
 
-    maximumPadding =
-      helpers.defaultNullOpts.mkInt 4
-        "Sets the maximum padding width with which to surround each tab";
+    highlight_visible = helpers.defaultNullOpts.mkBool true ''
+      Enables highlighting of visible buffers.
+    '';
+
+    icons = {
+      buffer_index =
+        helpers.defaultNullOpts.mkNullable
+          (
+            with types;
+            either bool (enum [
+              "superscript"
+              "subscript"
+            ])
+          )
+          false
+          "If `true`, show the index of the buffer with respect to the ordering of the buffers in the tabline.";
+
+      buffer_number = helpers.defaultNullOpts.mkNullable (
```

Comment:
```suggestion
      buffer_number = helpers.defaultNullOpts.mkNullableWithRaw (
```

Reply:
Maybe `defaultNullTypes` should get a `mkEither`...

Reply:
Yes why not.